### ISSUE TITLE
Variable input size training and data pooling

### ISSUE THREAD

**Issue by ziw-liu** [mehta-lab/VisCy#65]
To train a model with datasets of different dimensionalities (2D/3D) and imaging modalities (bright field, quantitative phase, Zernike phase contrast etc.), the training pipeline needs to have these new features:

- Data loading that draws samples from different data stores
  - Try the [combined dataloader](https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.utilities.combined_loader.html#lightning.pytorch.utilities.combined_loader.CombinedLoader.params.mode)
- Batch collation and training loop that works with 2D and 3D data at the same time
  - Use loss aggregation or alternation?
- A dynamic model stem with projection layers for different input dimensions
